### DISCUSSION TITLE
Criar cards com parte do código selecionado

### DISCUSSION THREAD

**Discussion by breakingtherabbit** [olmps/memo#21] [Ideias (Português)]
**Caso de Uso/Use Case:**

Selecionar parte do código para criação do _card_ de forma personalizada. 
Assim seria possível memorizar a parte do código realmente importante e não toda sua estrutura.

**Proposta/Proposal:**

Caso seja feita mais de uma seleção no código, criar a quantidades de _cards_ por seleção.
Por exemplo, o [mem.dev](https://mem.dev) utiliza-se desse recurso.

**Comment by matuella** [comment]
Olá @yzk23, antes de mais nada, muito obrigado pela contribuição.

Não sei se entendi muito bem a sua sugestão, mas a ideia é adicionar uma maneira de editar/criar o seu próprio card e permitir que o próprio usuário consiga inserir/remover/editar esse mesmo código?

Caso seja isso, eu acredito que é de fato uma ótima sugestão, no entanto, em primeira instância, iremos tentar melhorar um pouco mais da experiência da execução do card antes de permitir a criação de cards customizados. 

De qualquer maneira, vou deixar essa issue aqui aberta como ponto de melhoria futura, para quando voltarmos para estas definições de cards customizados.

**Comment by matuella** [comment]
Irei transformar esta issue em uma discussion de Idea, pois acredito ser o melhor lugar até construirmos uma ideia um pouco mais clara de como seria a implementação desta funcionalidade.

**Comment by breakingtherabbit** [comment]
Vou tentar explicar melhor. Vamos levar em consideração o _Insertion Sort_:
```c++
for (int j = 1; j < array_length; j++) {
    int key = array[j];
    int i = j - 1;

    while (i > -1 && array[i] > key) {
        array[i + 1] = array[i];
        i = i - 1;
    }

    array[i + 1] = key;
}
```
No cenário atual, me corrija se eu estiver errado, para eu criar um _flash card_ eu teria que criar um cartão para o algoritmo todo. 
Frente: O nome do algoritmo, por exemplo. E no verso: o código. 
Ou poderia escolher um trecho específico do algoritmo para criar um cartão, por exemplo, eu acho pertinente memorizar a condição ```i > -1 ``` do _while_  para  lembrar de limitar o algoritmo para não acessar uma posição que não existe do _array_.
Nesse caso eu colocaria na frente: limitação inferior do _Insertion Sort_, e no verso:  ```i > -1 ```.

O que estou proponto, como uma possível _feature_ futuramente. Eu poder inserir o algoritmo por completo e apenas selecionar o trecho que gostaria de memorizar. Assim se eu selecionasse mais de um trecho do próprio código seria tratado como dois cartões.
